drivers/metal/rendering_device_driver_metal.mm

```diff
@@ -141,28 +163,50 @@ _FORCE_INLINE_ MTLSize mipmapLevelSizeFromSize(MTLSize p_size, NSUInteger p_leve
 }
 
 void RenderingDeviceDriverMetal::buffer_free(BufferID p_buffer) {
-	rid::release(p_buffer);
+	BufferInfo *buf_info = (BufferInfo *)p_buffer.id;
+	buf_info->metal_buffer = 0; // Tell ARC to release.
```

Comment:
```suggestion
	buf_info->metal_buffer = nil; // Tell ARC to release.
```

Not necessary, as ARC will release the object when you free the C++ struct, but `nil` is the preferred token.